alica_engine/include/engine/blackboard/Blackboard.h

```diff
@@ -170,6 +170,20 @@ class BlackboardImpl
                 srcBb._vals.at(srcKey));
     }
 
+    void setConstValue(const std::string& targetKey, const std::string& value)
+    {
+        const auto yamlTypeIt = _yamlType.find(targetKey);
+        if (yamlTypeIt == _yamlType.end()) {
+            // key is not found in yaml file, invalid. Throw exception
+            throw BlackboardException(stringify("setConstValue() failure, unknown key used: ", targetKey, " is not defined in the yaml file"));
+        }
+        // key is found in yaml file
+        auto typeIndex = getTypeIndex(yamlTypeIt->second);
+        if (typeIndex.has_value()) {
+            _vals.at(targetKey) = makeBBValueForIndex<true>::make(typeIndex.value(), value);
+        }
```

Comment:
else throw exception. This means that yaml type is not known or is std::any. In either case we don't know how to parse it